### PR TITLE
Tl 285

### PR DESCRIPTION
**Pull request by humulla**
## Link to JIRA ticket

[Jira Ticket](https://hackney.atlassian.net/browse/TL-285)

## Describe this PR

Get By parentId implemented, this required to add a Global Secondary Index with parentId as the hash key

#### _Checklist_

- [x] Added end-to-end (i.e. integration) tests where necessary e.g to test complete functionality of a newly added feature
- [x] Added tests to cover all new production code
- [x] Checked all code for possible refactoring
- [x] Code pipeline builds correctly

### CODE REVIEW

File: PatchesApi.Tests/DynamoDbMockWebApplicationFactory.cs [hunk 1]
```diff
@@ -53,12 +54,29 @@ private static void EnsureTablesExist(IAmazonDynamoDB dynamoDb, List<TableDef> t
...
                 {
-                    var request = new CreateTableRequest(table.Name,
-                        new List<KeySchemaElement> { new KeySchemaElement(table.KeyName, KeyType.HASH) },
-                        new List<AttributeDefinition> { new AttributeDefinition(table.KeyName, table.KeyType) },
-                        new ProvisionedThroughput(3, 3));
-                    _ = dynamoDb.CreateTableAsync(request).GetAwaiter().GetResult();
+
+                    var keySchema = new List<KeySchemaElement> { new KeySchemaElement(table.KeyName, KeyType.HASH) };
+                    var attributes = new List<AttributeDefinition> { new AttributeDefinition(table.KeyName, table.KeyType) };
+                    //if (!string.IsNullOrEmpty(table.RangeKeyName))
```
Comment (evak6979) [inline PatchesApi.Tests/DynamoDbMockWebApplicationFactory.cs:60]: Comment can be removed.

File: PatchesApi.Tests/DynamoDbMockWebApplicationFactory.cs [hunk 2]
```diff
@@ -53,12 +54,29 @@ private static void EnsureTablesExist(IAmazonDynamoDB dynamoDb, List<TableDef> t
...
+                    //{
+                    //    keySchema.Add(new KeySchemaElement(table.RangeKeyName, KeyType.RANGE));
+                    //    attributes.Add(new AttributeDefinition(table.RangeKeyName, table.RangeKeyType));
+                    //}
+                    var indexKey = table.GlobalSecondaryIndexes.SelectMany(x => x.KeySchema).FirstOrDefault(y => y.KeyType == KeyType.HASH);
```
Comment (evak6979) [inline PatchesApi.Tests/DynamoDbMockWebApplicationFactory.cs:65]: Could change to first or default && keytype == whatever

File: PatchesApi.Tests/V1/Gateways/DynamoDbGatewayTests.cs [hunk 3]
```diff
@@ -88,10 +89,55 @@ public async Task GetPatchByIdReturnsThePatchIfItExists()
             _logger.VerifyExact(LogLevel.Debug, $"Calling IDynamoDBContext.LoadAsync for id parameter {query.Id}", Times.Once());
         }
 
+        [Fact]
+        public async Task GetByParentIdReturnsEmptyIfNoRecords()
+        {
+            var query = new GetPatchByParentIdQuery() { ParentId = Guid.NewGuid() };
+            Thread.Sleep(5000);
```
Comment (evak6979) [inline PatchesApi.Tests/V1/Gateways/DynamoDbGatewayTests.cs:96]: This can probably be removed now

File: PatchesApi/V1/Gateways/PatchesGateway.cs [hunk 4]
```diff
@@ -31,5 +36,47 @@ public async Task<PatchEntity> GetPatchByIdAsync(PatchesQueryObject query)
             var result = await _dynamoDbContext.LoadAsync<PatchesDb>(query.Id).ConfigureAwait(false);
             return result?.ToDomain();
         }
+
+        [LogCall]
+        public async Task<List<PatchEntity>> GetByParentIdAsync(GetPatchByParentIdQuery query)
+        {
+
```
Comment (evak6979) [inline PatchesApi/V1/Gateways/PatchesGateway.cs:43]: clean up newlines

File: PatchesApi.Tests/DynamoDbMockWebApplicationFactory.cs (hunk 2 above)
Comment (humulla) [reply PatchesApi.Tests/DynamoDbMockWebApplicationFactory.cs:65]: I don't think you can do that

**Review by evak6979 — APPROVED** [review]

**Pull request merged**